### PR TITLE
clang check performance-unnecessary-copy-initialization

### PR DESCRIPTION
**Pull request by davidlange6**
for comment about what clang-tidy does to CMSSW (and likely big and not mergeable - this is what clang-tidy check performance-unnecessary-copy-initialization does on the tip of CMSSW_9_3_X as of a few days ago

### CODE REVIEW

**Comment by davidlange6** [comment]
please test

**Comment by VinInn** [comment]
that's nice!

File: DataFormats/GeometrySurface/test/referencecounted_t.cpp [hunk 1]
```diff
@@ -51,7 +51,7 @@ namespace testreferencecounted {
       {
 	 RefPtr pointer( new RefTest );
 	 {
-	    RefPtr pointer2( pointer );
+	    const RefPtr& pointer2( pointer );
```
Comment (makortel) [inline DataFormats/GeometrySurface/test/referencecounted_t.cpp]: Intentional copy?

File: DataFormats/L1GlobalCaloTrigger/test/testL1GctInternHtMiss.cpp [hunk 2]
```diff
@@ -96,7 +96,7 @@ void testMethodsOfHtxVariant()
   assert(testObj2.hty() == 0);
   assert(testObj2.overflow() == false);
   
-  L1GctInternHtMiss testObj3(testObj1);  // Copy constructor
+  const L1GctInternHtMiss& testObj3(testObj1);  // Copy constructor
```
Comment (makortel) [inline DataFormats/L1GlobalCaloTrigger/test/testL1GctInternHtMiss.cpp]: Intentional copy?

File: DataFormats/L1GlobalCaloTrigger/test/testL1GctInternHtMiss.cpp [hunk 3]
```diff
@@ -133,7 +133,7 @@ void testMethodsOfHtyVariant()
   assert(testObj2.hty() == 7583);
   assert(testObj2.overflow() == false);
   
-  L1GctInternHtMiss testObj3(testObj1);  // Copy constructor
+  const L1GctInternHtMiss& testObj3(testObj1);  // Copy constructor
```
Comment (makortel) [inline DataFormats/L1GlobalCaloTrigger/test/testL1GctInternHtMiss.cpp]: Intentional copy?

File: DataFormats/L1GlobalCaloTrigger/test/testL1GctInternHtMiss.cpp [hunk 4]
```diff
@@ -170,7 +170,7 @@ void testMethodsOfHtxHtyVariant()
   assert(testObj2.hty() == -1714);
   assert(testObj2.overflow() == false);
   
-  L1GctInternHtMiss testObj3(testObj1);  // Copy constructor
+  const L1GctInternHtMiss& testObj3(testObj1);  // Copy constructor
```
Comment (makortel) [inline DataFormats/L1GlobalCaloTrigger/test/testL1GctInternHtMiss.cpp]: Intentional copy?

**Review by makortel — COMMENTED** [review]
I saw also other cases of likely intentional copies in unit tests, but didn't go through them all in detail.

**Comment by slava77** [comment]
-1

looks good for a starting point, but it looks like we cave a bunch of const-qualified modifiers which are not captured by this tool

**Pull request closed without merging**